src/Microsoft.PowerApps.TestEngine/TestEngine.cs

```diff
@@ -123,6 +123,18 @@ public async Task<string> RunTestAsync(FileInfo testConfigFile, string environme
             {
                 if (TestLoggerProvider.TestLoggers.ContainsKey(testRunId))
                 {
+                    if (!string.IsNullOrEmpty(testRunDirectory))
+                    {
+                        if (!Directory.Exists(testRunDirectory))
+                        {
+                            _fileSystem.CreateDirectory(testRunDirectory);
+                        }
+                    }
+                    else
+                    {
+                        testRunDirectory = "../../log";
```

Comment:
As discussed offline, we can take care of this separately.. for now just to facilitate this let's have the folders/files setup at the beginning of test execution so these pre-test failures are also logged gracefully.